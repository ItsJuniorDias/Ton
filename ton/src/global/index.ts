export * from './shadowStyles';
export * from './theme';
