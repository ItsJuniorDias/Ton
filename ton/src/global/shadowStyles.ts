export const shadowStyles = {
  shadowColor: '#000',
  shadowOffset: {
    width: 0,
    height: 5,
  },
  shadowOpacity: 0.1,
  shadowRadius: 2.62,
  elevation: 2,
};
