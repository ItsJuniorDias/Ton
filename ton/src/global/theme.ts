export const theme = {
  colors: {
    primary: '#01ad0b',
    secondary: '#01ea34',
    shape: '#ffffff',
    background: '#eff4f8',
    title: '#000000',
    text: '#20252b',
    notification: '#ff7b00',
    alert: '#ed5e69',
  },
  fonts: {
    regular: 'Roboto-Regular',
    medium: 'Roboto-Medium',
    bold: 'Roboto-Bold',
  },
};
