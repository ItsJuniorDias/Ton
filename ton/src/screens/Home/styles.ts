import styled from 'styled-components/native';
import { scale } from '../../utils';

export const Content = styled.View`
  flex: 1;
  padding: 0px 24px;
  padding-bottom: 32px;
  background-color: ${({ theme }) => theme.colors.background};
`;

export const Title = styled.Text`
  color: ${({ theme }) => theme.colors.primary};
  font-size: ${scale(18)};
  font-family: ${({ theme }) => theme.fonts.bold};
`;
