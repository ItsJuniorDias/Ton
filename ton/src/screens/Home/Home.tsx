import React from 'react';
import { FlatList, ActivityIndicator } from 'react-native';

import { Header, ProductCard } from '../../components';
import { useAppSelector, useFetchProducts } from '../../hooks';
import { theme } from '../../global';

import { Content, LoadingContent } from './styles';

export const Home = () => {
  const cart = useAppSelector((state) => state.cart);

  const {
    dataProducts: { data, loading },
  } = useFetchProducts();

  return (
    <>
      <Header title="Home" amountCart={cart.data.length} />

      {loading && (
        <LoadingContent>
          <ActivityIndicator size="large" color={theme.colors.secondary} />
        </LoadingContent>
      )}

      {!loading && (
        <Content>
          <FlatList
            data={data}
            numColumns={2}
            renderItem={({ item }) => <ProductCard {...item} />}
            keyExtractor={(item) => item.id}
          />
        </Content>
      )}
    </>
  );
};
