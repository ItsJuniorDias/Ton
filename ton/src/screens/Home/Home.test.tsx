import 'jest-styled-components';
import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { Home } from './Home';
import { theme } from '../../global';
import { Provider } from 'react-redux';
import store from '../../store';

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');

jest.mock('react-native-iphone-x-helper', () => ({
  getStatusBarHeight: jest.fn(),
  isIphoneX: jest.fn(),
}));

describe('Behavior Home', () => {
  const screenRender = () => (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <Home />
      </ThemeProvider>
    </Provider>
  );

  it('render snapshot', () => {
    const result = render(screenRender()).toJSON();

    expect(result).toMatchSnapshot();
  });
});
