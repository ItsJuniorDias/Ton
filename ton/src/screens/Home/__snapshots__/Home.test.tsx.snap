// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Behavior Home render snapshot 1`] = `
[
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "flexDirection": "row",
          "height": 126,
          "justifyContent": "space-between",
          "marginTop": "px",
          "paddingBottom": 0,
          "paddingLeft": 47,
          "paddingRight": 47,
          "paddingTop": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "height": 51,
            "width": 51,
          },
        ]
      }
    />
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Roboto-Medium",
            "fontSize": 35,
          },
        ]
      }
    >
      Header
    </Text>
    <View>
      <Icon
        color="#01ad0b"
        name="shopping-cart"
        size={28}
      />
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#ff7b00",
              "borderRadius": 20,
              "height": 35,
              "justifyContent": "center",
              "marginLeft": 31,
              "marginTop": -16,
              "position": "absolute",
              "width": 35,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "Roboto-Regular",
                "fontSize": 27,
              },
            ]
          }
          testID="amountCart_testID"
        >
          2
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      [
        {
          "backgroundColor": "#eff4f8",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 32,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 0,
        },
      ]
    }
  >
    <RCTScrollView
      data={
        [
          {
            "id": "T1",
            "price": 118.8,
            "thumbnail": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t1-1",
            "title": "T1 Brother",
          },
          {
            "id": "T1_CHIP",
            "price": 190.8,
            "thumbnail": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t1-chip-1",
            "title": "T1 Chip Brother",
          },
          {
            "id": "T2",
            "price": 310.8,
            "thumbnail": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t2-1",
            "title": "T2+ Brother",
          },
          {
            "id": "T3",
            "price": 418.8,
            "thumbnail": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t3-1",
            "title": "T3 Brother",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#ffffff",
                    "borderRadius": 31,
                    "height": 447,
                    "marginRight": 47,
                    "marginTop": 47,
                    "paddingBottom": 31,
                    "paddingLeft": 31,
                    "paddingRight": 31,
                    "paddingTop": 31,
                    "width": "46%",
                  },
                ]
              }
              testID="containerCard_testID"
            >
              <Image
                resizeMode="contain"
                source={
                  {
                    "uri": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t1-1",
                  }
                }
                style={
                  [
                    {
                      "height": 147,
                    },
                  ]
                }
              />
              <Text
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "#20252b",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 31,
                      "marginTop": 31,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                T1 Brother
              </Text>
              <View>
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "Roboto-Bold",
                        "fontSize": 24,
                        "lineHeight": 47,
                      },
                    ]
                  }
                >
                  R$ 118,80
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "Roboto-Bold",
                        "fontSize": 22,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontFamily": "Roboto-Bold",
                          "fontSize": 22,
                        },
                      ]
                    }
                  >
                    12x 
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "color": "#01ad0b",
                          "fontFamily": "Roboto-Bold",
                          "fontSize": 35,
                        },
                      ]
                    }
                  >
                    R$ 9,90
                  </Text>
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "backgroundColor": "#01ea34",
                      "borderRadius": 31,
                      "height": 51,
                      "justifyContent": "center",
                      "marginTop": 31,
                      "width": "100%",
                    },
                    {
                      "opacity": 1,
                    },
                  ]
                }
                testID="button_testID"
              >
                <Text
                  style={
                    [
                      {
                        "color": "#20252b",
                        "fontFamily": "Roboto-Bold",
                        "fontSize": 27,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Pedir agora
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "#ffffff",
                    "borderRadius": 31,
                    "height": 447,
                    "marginRight": 47,
                    "marginTop": 47,
                    "paddingBottom": 31,
                    "paddingLeft": 31,
                    "paddingRight": 31,
                    "paddingTop": 31,
                    "width": "46%",
                  },
                ]
              }
              testID="containerCard_testID"
            >
              <Image
                resizeMode="contain"
                source={
                  {
                    "uri": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t1-chip-1",
                  }
                }
                style={
                  [
                    {
                      "height": 147,
                    },
                  ]
                }
              />
              <Text
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "#20252b",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 31,
                      "marginTop": 31,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                T1 Chip Brother
              </Text>
              <View>
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "Roboto-Bold",
                        "fontSize": 24,
                        "lineHeight": 47,
                      },
                    ]
                  }
                >
                  R$ 190,80
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "Roboto-Bold",
                        "fontSize": 22,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontFamily": "Roboto-Bold",
                          "fontSize": 22,
                        },
                      ]
                    }
                  >
                    12x 
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "color": "#01ad0b",
                          "fontFamily": "Roboto-Bold",
                          "fontSize": 35,
                        },
                      ]
                    }
                  >
                    R$ 15,90
                  </Text>
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "backgroundColor": "#01ea34",
                      "borderRadius": 31,
                      "height": 51,
                      "justifyContent": "center",
                      "marginTop": 31,
                      "width": "100%",
                    },
                    {
                      "opacity": 1,
                    },
                  ]
                }
                testID="button_testID"
              >
                <Text
                  style={
                    [
                      {
                        "color": "#20252b",
                        "fontFamily": "Roboto-Bold",
                        "fontSize": 27,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Pedir agora
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#ffffff",
                    "borderRadius": 31,
                    "height": 447,
                    "marginRight": 47,
                    "marginTop": 47,
                    "paddingBottom": 31,
                    "paddingLeft": 31,
                    "paddingRight": 31,
                    "paddingTop": 31,
                    "width": "46%",
                  },
                ]
              }
              testID="containerCard_testID"
            >
              <Image
                resizeMode="contain"
                source={
                  {
                    "uri": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t2-1",
                  }
                }
                style={
                  [
                    {
                      "height": 147,
                    },
                  ]
                }
              />
              <Text
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "#20252b",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 31,
                      "marginTop": 31,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                T2+ Brother
              </Text>
              <View>
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "Roboto-Bold",
                        "fontSize": 24,
                        "lineHeight": 47,
                      },
                    ]
                  }
                >
                  R$ 310,80
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "Roboto-Bold",
                        "fontSize": 22,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontFamily": "Roboto-Bold",
                          "fontSize": 22,
                        },
                      ]
                    }
                  >
                    12x 
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "color": "#01ad0b",
                          "fontFamily": "Roboto-Bold",
                          "fontSize": 35,
                        },
                      ]
                    }
                  >
                    R$ 25,90
                  </Text>
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "backgroundColor": "#01ea34",
                      "borderRadius": 31,
                      "height": 51,
                      "justifyContent": "center",
                      "marginTop": 31,
                      "width": "100%",
                    },
                    {
                      "opacity": 1,
                    },
                  ]
                }
                testID="button_testID"
              >
                <Text
                  style={
                    [
                      {
                        "color": "#20252b",
                        "fontFamily": "Roboto-Bold",
                        "fontSize": 27,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Pedir agora
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "#ffffff",
                    "borderRadius": 31,
                    "height": 447,
                    "marginRight": 47,
                    "marginTop": 47,
                    "paddingBottom": 31,
                    "paddingLeft": 31,
                    "paddingRight": 31,
                    "paddingTop": 31,
                    "width": "46%",
                  },
                ]
              }
              testID="containerCard_testID"
            >
              <Image
                resizeMode="contain"
                source={
                  {
                    "uri": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t3-1",
                  }
                }
                style={
                  [
                    {
                      "height": 147,
                    },
                  ]
                }
              />
              <Text
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "#20252b",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 31,
                      "marginTop": 31,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                T3 Brother
              </Text>
              <View>
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "Roboto-Bold",
                        "fontSize": 24,
                        "lineHeight": 47,
                      },
                    ]
                  }
                >
                  R$ 418,80
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "Roboto-Bold",
                        "fontSize": 22,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontFamily": "Roboto-Bold",
                          "fontSize": 22,
                        },
                      ]
                    }
                  >
                    12x 
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "color": "#01ad0b",
                          "fontFamily": "Roboto-Bold",
                          "fontSize": 35,
                        },
                      ]
                    }
                  >
                    R$ 34,90
                  </Text>
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "backgroundColor": "#01ea34",
                      "borderRadius": 31,
                      "height": 51,
                      "justifyContent": "center",
                      "marginTop": 31,
                      "width": "100%",
                    },
                    {
                      "opacity": 1,
                    },
                  ]
                }
                testID="button_testID"
              >
                <Text
                  style={
                    [
                      {
                        "color": "#20252b",
                        "fontFamily": "Roboto-Bold",
                        "fontSize": 27,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Pedir agora
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>,
]
`;
